src/foam/nanos/auth/UserAndGroupAuthService.java

```diff
@@ -167,6 +167,46 @@ public User loginByEmail(X x, String email, String password) throws Authenticati
     return user;
   }
 
+  /**
+    Checks if the user passed into the method has the passed
+    in permission attributed to it by checking their group.
+    No check on User and group enabled flags.
+  */
+  public Boolean checkUserPermission(foam.core.X x, User user, String permission) {
```

Comment:
return type should be 'boolean' not Boolean.
Fix the other two check() methods while you're at it.